packages/solutions/app-tools/src/index.ts

```diff
@@ -152,7 +152,21 @@ export const appTools = (
       async watchFiles() {
         const appContext = api.useAppContext();
         const config = api.useResolvedConfigContext();
-        return await generateWatchFiles(appContext, config.source.configDir);
+        const files = await generateWatchFiles(
+          appContext,
+          config.source.configDir,
+        );
+
+        const watchFiles = config.dev.watchFiles;
+        if (watchFiles?.type === 'reload-server') {
```

Comment:
`watchFiles` might be an array in the latest version of Rsbuild
https://github.com/web-infra-dev/rsbuild/pull/3640